src/RequestHandler/inprocess/inprocessapplication.cpp

```diff
@@ -544,27 +562,82 @@ IN_PROCESS_APPLICATION::ExecuteApplication(
     // set the callbacks
     s_Application = this;
 
-    RunDotnetApplication(m_pConfig->QueryHostFxrArgCount(), m_pConfig->QueryHostFxrArguments(), pProc);
+    hr = RunDotnetApplication(m_pConfig->QueryHostFxrArgCount(), m_pConfig->QueryHostFxrArguments(), pProc);
 
 Finished:
     //
     // this method is called by the background thread and should never exit unless shutdown
     //
-    if (!m_fRecycleProcessCalled)
+    if (!m_fRecycleProcessCalled && FAILED(hr))
     {
-        STRU                    strEventMsg;
+        // Close the pipe as Read will hang if it isn't set.
+        CHAR pzFileContents[4096] = { 0 };
+        DWORD dwNumBytesRead;
+        STRU  struStdErrLog;
+        LARGE_INTEGER   li = { 0 };
+
+        // Try to read std err logs for application exception.
+        // If anything fails here, ignore it.
+
+        if (m_pConfig->QueryStdoutLogEnabled())
+        {
+            if (m_hLogFileHandle != INVALID_HANDLE_VALUE)
+            {
+                if (!GetFileSizeEx(m_hLogFileHandle, &li))
```

Comment:
do we care about the failure here?

Reply:
We don't (can't do anything about it). I'll just check that both calls succeeded before trying to read.